src/libraries/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -1460,22 +1452,17 @@ public static void Reverse(Array array, int index, int length)
                     return;
                 case CorElementType.ELEMENT_TYPE_I4:
                 case CorElementType.ELEMENT_TYPE_U4:
-#if TARGET_32BIT
```

Comment:
It was done this way to reduce number of generic instantations that this needs. This is undoing that optimization.